src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventFilter.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.event;
+
+import static uk.gov.ons.ctp.common.log.ScopedStructuredArguments.kv;
+
+import java.util.Set;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.Survey;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CollectionExerciseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.SurveyRepository;
+
+@Slf4j
+@Component
+public class EventFilter {
+
+  private final AppConfig appConfig;
+  private final SurveyRepository surveyRepository;
+  private final CollectionExerciseRepository collectionExerciseRepository;
+
+  public EventFilter(
+      AppConfig appConfig,
+      SurveyRepository surveyRepository,
+      CollectionExerciseRepository collectionExerciseRepository) {
+    this.appConfig = appConfig;
+    this.surveyRepository = surveyRepository;
+    this.collectionExerciseRepository = collectionExerciseRepository;
+  }
+
+  public boolean isValidEvent(String surveyId, String collexId, String caseId, String messageId) {
+
+    log.info("Entering acceptCaseEvent {}, {}", kv("messageId", messageId), kv("caseId", caseId));
+
+    Survey survey = findSurvey(surveyId);
+    boolean acceptedSurveyType = isAcceptedSurveyType(survey);
+    boolean valid = acceptedSurveyType && isKnownCollectionExercise(collexId);
```

Comment:
We cannot store the case in the DB without also having the collexid due to the FK constraint, but...
If the case is for a valid _survey_ BUT we don't have the collex YET, does not mean we should throw away the case
We will want to throw the Case back to the subscription / ex manager when it exists, until such time that we get the collex sent to us.
If the survey is OK but we do not have the collex yet, throw a CTPException and let it go back to pub sub

Reply:
Done plus same for missing survey as discussed